src/MessagePack.CodeGenerator/CodeAnalysis/TypeCollector.cs

```diff
@@ -626,13 +626,18 @@ void CollectObject(INamedTypeSymbol type)
             }
 
             // GetConstructor
+            IEnumerator<IMethodSymbol> ctorEnumerator = null;
             var ctor = type.Constructors.Where(x => x.DeclaredAccessibility == Accessibility.Public).SingleOrDefault(x => x.GetAttributes().Any(y => y.AttributeClass == typeReferences.SerializationConstructorAttribute));
```

Comment:
There are two ways to generate dynamic code.
AssemblyBuilder and DynamicMethod.
AssemblyBuilder can not access private.
And MessagePack-CSharp usually adopts AssemblyBuilder.
AssemblyBuilder has various advantages, it is currently faster than DynamicMethod.